servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/RetryingHttpRequesterFilterBuilder.java

```diff
@@ -0,0 +1,126 @@
+     * retry for requests. This behavior may add latency to requests till the time the load balancer is ready
+     * instead of failing fast. This method allows controlling that behavior.
+     *
+     * @param waitForLb Whether to wait for the {@link LoadBalancer} to be ready before retrying requests.
+     * @return {@code this}.
+     */
+    public RetryingHttpRequesterFilterBuilder waitForLoadBalancer(final boolean waitForLb) {
+        this.waitForLb = waitForLb;
+        return this;
+    }
+
+    /**
+     * By default, fail a request if the last signal from the associated {@link ServiceDiscoverer} was an error.
+     * This method disables that behavior.
+     *
+     * @param ignoreSdErrors ignore {@link ServiceDiscoverer} errors when evaluating a request failure.
+     * @return {@code this}.
+     */
+    public RetryingHttpRequesterFilterBuilder ignoreServiceDiscovererErrors(final boolean ignoreSdErrors) {
+        this.ignoreSdErrors = ignoreSdErrors;
+        return this;
+    }
+
+    /**
+     * Set the maximum number of allowed retry operations before giving up, applied as total max across both
+     * {@link #retryRequests(BiFunction)} and {@link #retryResponses(Function)}.
+     *
+     * @param maxRetries Maximum number of allowed retries before giving up
+     * @return {@code this}
+     */
+    public RetryingHttpRequesterFilterBuilder maxTotalRetries(final int maxRetries) {
+        if (maxRetries <= 0) {
+            throw new IllegalArgumentException("maxRetries: " + maxRetries + " (expected: >0)");
+        }
+        this.maxRetries = maxRetries;
+        return this;
+    }
+
+    /**
+     * Overrides the default criterion for determining which responses should be retried.
+     * <strong>It's important that this {@link Function} doesn't block to avoid performance impacts.</strong>
+     *
+     * @param mapper {@link Function} that checks whether a given {@link HttpResponseMetaData meta-data} should
+     * be retried, producing a {@link BackOffPolicy} in such cases.
+     * @return {@code this}
+     */
+    public RetryingHttpRequesterFilterBuilder retryResponses(
+            final Function<HttpResponseMetaData, BackOffPolicy> mapper) {
+        this.retryForResponsesMapper = requireNonNull(mapper);
+        return this;
+    }
+
+    /**
+     * Overrides the default criterion for determining which requests or errors should be retried.
+     * <strong>It's important that this {@link Function} doesn't block to avoid performance impacts.</strong>
+     *
+     * @param mapper {@link BiFunction} that checks whether a given combination of
+     * {@link HttpRequestMetaData meta-data} and {@link Throwable cause} should be retried, producing a
+     * {@link BackOffPolicy} in such cases.
+     * @return {@code this}
+     */
+    public RetryingHttpRequesterFilterBuilder retryRequests(
+            final BiFunction<HttpRequestMetaData, Throwable, BackOffPolicy> mapper) {
```

Comment:
Always requiring to return a `BackOffPolicy` looks like an overkill. Did you consider keeping both: a `BiPredicate` and `BiFunction`? Then users may choose `BiPredicate` if they don't need to worry about different `BackOffPolicies`. A builder method that takes `BiPredicate` may take `BackOffPolicy` as a 2nd argument.